transcom/mymove
# B 21168 int Zip/City/State Finder

## [B-21168](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1011663)

## Summary

This PR creates a new Zip/City/State search component that will be used to search and populate the City State Zip and County fields for the Customer Profile Addresses and the various Office app pages and modals. The fields are being made read only to prevent users from entering bad data into the Address Fields.
An API was added to use to return the results of the search and model was added to use the new v_locations view to query against. 

Several pages were modified in the Office App and the Customer Profile creation was updated. The rest of the Customer pages will be updated as part of B-21169 and the Prime Sim will be updated in another backlog.

Is there anything you would like reviewers to give additional scrutiny?

There are many event handler methods for setting the Formik values for the City, State, County, Zip fields with the only difference being the object used to store the values. If anybody knows of a way to access an object using the name of it passed in as a string I would appreciate some help. This would reduce the multiple event handler methods down to one per file. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Make sure you can search by just the Postal Code, or by using City, State Postal Code (Note the , after City that is required to help distinguish between City and State). Make sure you can save and continue once you select a result and fill out the rest of the pages. 

NOTE: The City State, Zip and County fields should all be read only when the location lookup component is present. 

1. Access the Customer app and create a new customer profile
2. Test the Customer Address and the Backup Address pages 
3. Login to the Office App and make sure you can edit customer profile address, edit shipment addresses using the location lookup tool. Try the various shipment types and edit them or add a new shipment and test the location lookup.

### Frontend


### Backend


## Screenshots
